lib/index.js

```diff
@@ -107,7 +107,11 @@ var flushing = false;
 
 function flushUpdate() {
   flushing = true;
-  while (toUpdate.length > 0) updateDeps(toUpdate.shift());
+  while (toUpdate.length > 0) {
+    var s = toUpdate.shift();
+    if (s.vals.length > 0) s.val = s.vals.shift();
+    updateDeps(s);
+  }
```

Comment:
Yeah that was my first thought too but realised that would require too much of a code change.